Documentation/UPI/vSphere_terraform/README.md

```diff
@@ -0,0 +1,201 @@
+
+## Prepare environment
+
+Checkout the official [Openshift 4.3 Documentation](https://docs.openshift.com/container-platform/4.3/installing/installing_vsphere/installing-vsphere.html#installation-requirements-user-infra_installing-vsphere). Add DNS and DHCP records for your virtual machines, that will be provisioned in next steps.
+
+Install a Loadbalancer in front of your API servers. Choose a high availability setup, if you need it. The Loadbalancer is only for API traffic and Kubernetes can handle an outage of the API. During an outtage the cluster is not able to change something.
+
+Find, download and upload an image of FCOS for [VMware vSphere](https://getfedora.org/en/coreos/download/).
+
+### Example Environment
+
+This example environment includes three master and worker nodes. Etcd is running on master nodes.
+
+#### A-records with mac address
+
+| FQDN | A-record | MAC address |
+|------|----------|-------------|
+| bootstrap.okd.example.com | 10.20.15.3 | 00:1c:14:00:00:03 |
+| lb.okd.example.com | 10.20.15.2 | 00:1c:14:00:00:02 |
+| api.okd.example.com | 10.20.15.2 | |
+| api-int.okd.example.com | 10.20.15.2 | |
+| master1.okd.example.com | 10.20.15.11 | 00:1c:14:00:00:11 |
+| master2.okd.example.com | 10.20.15.12 | 00:1c:14:00:00:12 |
+| master3.okd.example.com | 10.20.15.13 | 00:1c:14:00:00:13 |
+| etcd-0.okd.example.com | 10.20.15.11 | |
+| etcd-1.okd.example.com | 10.20.15.12 | |
+| etcd-2.okd.example.com | 10.20.15.13 | |
+| worker1.okd.example.com | 10.20.15.41 | 00:1c:14:00:00:41 |
+| worker2.okd.example.com | 10.20.15.42 | 00:1c:14:00:00:42 |
+| worker3.okd.example.com | 10.20.15.43 | 00:1c:14:00:00:43 |
+
+#### SRV-records
+
+| domain | TTL | priority | weight | port | target |
+|--------|-----|----------|--------|------|--------|
+| _etcd-server-ssl._tcp.okd.example.com | 86400 | 0 | 10 | 2380 (etcd) | 10.20.15.11, 10.20.15.12, 10.20.15.13
+
+#### wildcard DNS record
+
+| domain | A-record | comment |
+|--------|----------|---------|
+| *.apps.okd.example.com | 10.20.15.41, 10.20.15.42, 10.20.15.43 | For high availability, we choose the first three worker nodes, which should run the router component |
+
+#### Loadbalancer - HAProxy configuration
+
+Note: Binding the API port also to 443 allows us to connect easyly with `oc`, no need for port number, e.g. `oc login api.okd.example.com`
+
+```HAProxy
+global
+    log         127.0.0.1 local2
+
+    chroot      /var/lib/haproxy
+    pidfile     /var/run/haproxy.pid
+    maxconn     4000
+    user        haproxy
+    group       haproxy
+    daemon
+
+    stats socket /var/lib/haproxy/stats
+
+    ssl-default-bind-ciphers PROFILE=SYSTEM
+    ssl-default-server-ciphers PROFILE=SYSTEM
+
+defaults
+    mode                    http
+    log                     global
+    option                  httplog
+    option                  dontlognull
+    timeout connect         10s
+    timeout client          1m
+    timeout server          1m
+
+listen  stats
+    bind 10.20.15.2:9000
```

Comment:
Lets add the comment that its loadbalancer address (a bit tough to navigate in the ips)

Reply:
Changed to `*:9000` as the other frontends.